### PR TITLE
Back port some stable patches for v0.5.x

### PR DESCRIPTION
**Pull request by changweige**
Clearly pick some stable patches for v0.5 and prepare to release v0.5.1

### CODE REVIEW

File: pkg/manager/manager.go [hunk 1]
```diff
@@ -311,8 +311,9 @@ func (m *Manager) handleDaemonDeathEvent() {
 			log.L.Warnf("Daemon %s was not found", ev.daemonID)
 			return
 		}
-		collector.NewDaemonInfoCollector(&d.Version, -1).Collect()
+
 		d.Lock()
+		collector.NewDaemonInfoCollector(&d.Version, -1).Collect()
```
Comment (sctb512) [inline pkg/manager/manager.go:316]: Whether is a lock needed before another call to this method?

Comment (sctb512) [reply pkg/manager/manager.go:316]: In a few cases, I get the extra values that `d.version` is incorrect. However, the value of the expected version is eventually correct.

Comment (changweige) [reply pkg/manager/manager.go:316]: What do you mean by "incorrect"?  The version string is shorter than expected or polluted?

Comment (sctb512) [reply pkg/manager/manager.go:316]: The `version` string is null. It appeared by chance.

**Pull request merged**